### ISSUE TITLE
Light/Dark theme option and mirroring system setting

### ISSUE THREAD

**Issue by nazriel** [zellij-org/zellij#3277]
Hello,

I am using Light system theme during the day (since it "energizes" me more) and then dark system theme during the evening and night (since it saves my eyes and helps with winding down). OS automatically toggles between them depending on the time of the day. 

Right now most of my apps follow this by default or with small help (like f-person/auto-dark-mode.nvim for nvim and little lua snippet for wezterm). 

I was wondering if there is already working solution for this with Zellij?
Either by default config (ie. akin to how Zed does it [1]), plugin oraz 3rd party solution?

[1] 

```
"theme": {
  "mode": "light" | "dark" | "system",
  "dark": "One Dark",
  "light": "One Light"
},

```

**Comment by mikaoelitiana** [comment]
This is also a feature I am really looking forward to having

**Comment by nazriel** [comment]
From the likes and @mikaoelitiana comment I will assume that I am not blind and such functionality is not available yet.
I will give it a try to implement it and share it here

**Comment by nazriel** [comment]
I was thinking something like that: 

https://github.com/zellij-org/zellij/compare/main...nazriel:zellij:nazriel/dark-light-theme

Idea was following:
1. Introducing new config options: theme_dark, theme_light
2. During client startup we detect current system theme and see if corresponding  theme_dark/theme_light is set - if yes we create palette based on theme set in theme_dark/theme_light variable (already works)
3. In client we poll and check if system theme changed and if so we send message to server to resync client_attributes with new palette set (still WIP)

**Comment by cristiand391** [comment]
IMO, config live-reload is enough to allow people script this in any OS/distribution.
I would prefer to setup mac/linux to run a script that updates the current theme in my config.kdl than have the zellij server to constantly poll for system theme changes.

**Comment by TheButlah** [comment]
If its a helpful datapoint, I would prefer not to have multiple config files for zellij, with one per theme. Instead, I think having the concept of multiple themes defined in the config makes more sense. You can then in your config file designate one as the default dark mode theme, one as the default light mode theme, and then configure whether the light mode is dark, light, or automatic by default. 

You can also have a command to control the mode, to enable people to script this. But I do think that automatic should be one of the options to use for modes, for those of us who just want a good all in one system without additional scripting.

**Comment by gregorias** [comment]
To address this feature gap, I change the theme in the config whenever I switch between light or dark modes like so:

```fish
function _change_zellij_theme -a THEME
        set -Ux ZELLIJ_CURRENT_THEME $THEME
        sed -i '' "s/^\(theme \)\(.*\)\$/\1\"$ZELLIJ_CURRENT_THEME\"/" $XDG_CONFIG_HOME/zellij/config.kdl
end
```

It works, because Zellij watches and reloads based on config changes. This function is used in my wider setup that monitors the OS’ light/dark settings.

**Comment by cristiand391** [comment]
to add to @gregorias's comment, live reloading was added in zellij 0.41.0:
https://zellij.dev/news/colliding-keybinds-plugin-manager/

I see there's also a new `reconfigure` plugin command:
https://zellij.dev/documentation/plugin-api-commands#reconfigure

would anyone be interested in a headless plugin that you can just send a msg through pipes to change the theme? like a headless plugin that's running in the background, then you could run something like this in scripts that run when switching between light/dark:
```
zellij pipe --plugin zj-theme -- gruvbox-dark
```

could also work with keybinds

**Comment by jonathanslenders** [comment]
Related: https://github.com/zellij-org/zellij/issues/3831

**Comment by sarrietav-dev** [comment]
I did this in gnome, it doesn't automatically change the theme, but it is something!

```bash
zz() {
  theme=$(gsettings get org.gnome.desktop.interface color-scheme)
  if [[ $theme == *"prefer-dark"* ]]; then
    zellij options --theme gruvbox-dark
  else
    zellij options --theme gruvbox-light
  fi
}
```

**Comment by raszi** [comment]
I am considering switching from tmux to Zellij, and this is a feature that I definitely miss. Tried multiple themes, but none of them really worked in a light setup that works well in dark and vice versa.

**Comment by TheButlah** [comment]
Mutating the config with sed + scripting is not a good solution. And it doesn't even work on distros that are immutable like NixOS.

**Comment by jaysonsantos** [comment]
hey folks, i think this issue should be closed in favor of https://github.com/zellij-org/zellij/issues/3831 which seems to be the way other projects are adopting this switch

**Comment by mikaoelitiana** [comment]
> to add to [@gregorias](https://github.com/gregorias)'s comment, live reloading was added in zellij 0.41.0: https://zellij.dev/news/colliding-keybinds-plugin-manager/
> 
> I see there's also a new `reconfigure` plugin command: https://zellij.dev/documentation/plugin-api-commands#reconfigure
> 
> would anyone be interested in a headless plugin that you can just send a msg through pipes to change the theme? like a headless plugin that's running in the background, then you could run something like this in scripts that run when switching between light/dark:
> 
> ```
> zellij pipe --plugin zj-theme -- gruvbox-dark
> ```
> 
> could also work with keybinds

@cristiand391 I am interested in something like this. Have you created something like this or know if it already exists?

**Comment by jokeyrhyme** [comment]
one workaround not mentioned here so far is to use the "ansi" theme 

then the light/dark concern is lifted completely out of zellij, because it will use whatever colours are provided by the terminal emulator

then you can configure your terminal emulator to switch between light/dark as appropriate (assuming it has this feature)

**Comment by zetashift** [comment]
> one workaround not mentioned here so far is to use the "ansi" theme
> 
> then the light/dark concern is lifted completely out of zellij, because it will use whatever colours are provided by the terminal emulator
> 
> then you can configure your terminal emulator to switch between light/dark as appropriate (assuming it has this feature)

This workaround is the one I currently use, but what would make proper CSI 2031 support(or passthrough?) enticing is that other terminal apps like helix will automaticaly switch to the dark/light theme. And the themes themselves have a lot more (semantic) coloring than ansi.

**Comment by mikaoelitiana** [comment]
> one workaround not mentioned here so far is to use the "ansi" theme
> 
> then the light/dark concern is lifted completely out of zellij, because it will use whatever colours are provided by the terminal emulator
> 
> then you can configure your terminal emulator to switch between light/dark as appropriate (assuming it has this feature)

I don't know if I am missing something but I can't make this work using Ghostty. Zellij would always keep dark theme and by the way, `ansi` is listed in [Dark Themes on the doc](https://zellij.dev/documentation/theme-list.html#dark-themes). 

![Image](https://github.com/user-attachments/assets/d2b1ebd7-371a-4070-b715-d1906fad40cd)

**Comment by Mach1212** [comment]
Yeah it doesn't work for me either @mikaoelitiana. I also use ghostty and no matter what it stays dark mode unless I explicitly set a light theme.

```ghostty
confirm-close-surface = true
theme = light:Catppuccin Latte, dark:Catppuccin Macchiato
window-theme = system
```
```zellij
theme "default"
theme_dark "catppuccin-macchiato"
theme_light "catppuccin-latte"
```

**Comment by imsnif** [comment]
Implemented in 0.44.2.

**Comment by Mach1212** [comment]
@imsnif I am on 44.2 

<img width="227" height="67" alt="Image" src="https://github.com/user-attachments/assets/3b32c18d-9d7d-434e-98e4-e50f2b605a22" />

it doesn't work or give me an example config that does?
My neovim receives the change and so does ghostty, but not zellij...

**Comment by charego** [comment]
Thanks @imsnif! Quick question on usage after reading the release notes and trying it out.

Say my system is on light theme and my config is:

```
theme "catppuccin-frappe"
theme_dark "catppuccin-frappe"
theme_light "catppuccin-latte"
```

When I open a new zellij instance, it starts with catppuccin-frappe (`theme`), a dark theme.

Would it be possible to automatically choose the theme matching the system - `theme_light` in this example?

I wonder if `theme` could be ignored when `theme_dark`/`theme_light` are set.

**Comment by charego** [comment]
Also, not to pile on, but another observation: when I change the system theme after opening zellij, the "startup tips" modal doesn't react to the theme change. I need to nudge it to force a redraw, e.g. by cycling through the tips with the arrow keys. Both of these issues can be worked around by setting `theme` to the system theme I use most often.

**Comment by jokeyrhyme** [comment]
> I wonder if theme could be ignored when theme_dark/theme_light are set.

i'm planning to set theme=ansi and then theme_dark/theme_light to my desired themes, so that zellij at least matches my terminal emulator in the absence of specific signals either way

**Comment by lihsai0** [comment]
<img width="1452" height="630" alt="Image" src="https://github.com/user-attachments/assets/d5dad7bf-2e5e-4465-ad0f-c91bf9c72816" />

Both `theme_dark` and `theme_light` have been configured, but not working.

I must be missing somethings.

---

After deleting the old config file to make Zellij regenerate it, everything working great.
